File: examples/java-calculator/src/main/java/io/cucumber/examples/java/DateCalculator.java
package io.cucumber.examples.java;

import java.time.LocalDate;

public class DateCalculator {

    private final LocalDate now;

    public DateCalculator(LocalDate now) {
        this.now = now;
    }

    public String isDateInThePast(LocalDate date) {
        return (date.isBefore(now)) ? "yes" : "no";
    }

}

File: examples/calculator-kotlin-junit5/src/main/kotlin/io/cucumber/examples/calaculator/DateCalculator.kt
package io.cucumber.examples.calaculator

import java.time.LocalDate
import java.util.*

class DateCalculator(private val now: LocalDate) {

    fun isDateInThePast(date: LocalDate): String {
        return if (date.isBefore(now)) "yes" else "no"
    }
}

